addons/l10n_tr_nilvera_edespatch/models/res_partner.py

```diff
@@ -0,0 +1,31 @@
+from odoo import _, models
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    def _l10n_tr_nilvera_validate_partner_details(self):
+        error_messages = {}
+        for record in self:
+            required_fields = {
+                "street": record.street,
+                "city": record.city,
+                "state": record.state_id,
+                "zip": record.zip,
+                "country": record.country_id,
+            }
+            missing_fields = [field.capitalize() for field, value in required_fields.items() if not value]
+            if record.country_id.code == 'TR' and not record.vat:
+                missing_fields.append("TCKN/VKN")
+
+            msg = []
+            if missing_fields:
+                msg.append(f"{', '.join(missing_fields)} is required")
+            if record.zip and len(record.zip) != 5:
+                msg.append("ZIP must be of 5 characters.")
```

Comment:
```suggestion
                msg.append(_("ZIP must be of 5 characters."))
```
Make sure to re-export pot after any changes while any changes to translation strings